packages/cherry-pick-bot/src/cherry-pick.ts

```diff
@@ -0,0 +1,355 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {logger} from 'gcf-utils';
+// eslint-disable-next-line node/no-extraneous-import
+import {Octokit} from '@octokit/rest';
+// eslint-disable-next-line node/no-extraneous-import
+import {RequestError} from '@octokit/request-error';
+import * as crypto from 'crypto';
+
+type OctokitType = InstanceType<typeof Octokit>;
+
+interface PullRequest {
+  number: number;
+  html_url: string;
+  title: string;
+  body: string | null;
+}
+
+const COMMENT_REGEX = /^\/cherry-pick\s+(?<branch>\w[-\w]*)/;
+
+/**
+ * Parse a comment string to see if it matches the expected cherry-pick
+ * command of `/cherry-pick <branch-name>`
+ *
+ * @param {string} comment The comment to parse
+ * @returns {string|null} Returns the parsed branch name to cherry-pick to
+ *   or null if it does not match
+ */
+export function parseCherryPickComment(comment: string): string | null {
+  const match = comment.trim().match(COMMENT_REGEX);
+  if (match?.groups) {
```

Comment:
nit: I think you could just do `return match?.groups?.branch`, if you allowed for `undefined` return rather than `null`.